contracts/StablecoinConverter.sol

```diff
@@ -108,4 +109,63 @@ contract StablecoinConverter is EpochTokenLocker {
     function tokenIdToAddressMap(uint16 id) public view returns (address) {
         return IdToAddressBiMap.getAddressAt(registeredTokens, id);
     }
-}
\ No newline at end of file
+
+    struct TradeData {
+        address owner;
+        uint volume;
+        uint16 orderIds;
+    }
+
+    function submitSolution(
+        uint32 batchIndex,
+        address[] memory owners,  //tradeData is submitted as arrays
+        uint16[] memory orderIds,
+        uint128[] memory volumes,
+        uint128[] memory prices,  //list of prices for touched token only
+        uint16[] memory tokenIdsForPrice  // price[i] is the price for the token with tokenID tokenIdsForPrice[i]
+    ) public {
+        require(
+            batchIndex == getCurrentStateIndex() - 1,
+            "Solutions are no longer accepted for this batch"
+        );
+        uint len = owners.length;
+        for (uint i = 0; i < len; i++) {
+            Order memory order = orders[owners[i]][orderIds[i]];
+            require(order.validFrom <= batchIndex, "Order is not yet valid");
+            require(order.validUntil >= batchIndex, "Order is no longer valid");
+            // Assume for now that we always have sellOrders
+            uint128 executedSellAmount = volumes[i];
+            uint128 executedBuyAmount = uint128(
+                volumes[i].mul(prices[findPriceIndex(order.buyToken, tokenIdsForPrice)]) /
+                prices[findPriceIndex(order.sellToken, tokenIdsForPrice)]
```

Comment:
Given you will be writing the price list in SC storage anyways (in the next PR), would it make sense to store it as a mapping on storage and then access it in O(1) instead of doing a linear search over it?